riot/lol/challenges.go

```diff
@@ -0,0 +1,81 @@
+package lol
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"github.com/yigithanbalci/golio/internal"
+)
+
+type ChallengesClient struct {
+	c *internal.Client
+}
+
+func (cc *ChallengesClient) GetConfig() ([]*ChallengeConfigInfoDto, error) {
+	logger := cc.logger().WithField("method", "GetConfig")
+	var challengeConfigs []*ChallengeConfigInfoDto
+	if err := cc.c.GetInto(endpointChallangesConfig, &challengeConfigs); err != nil {
+		logger.Debug(err)
+		return nil, err
+	}
+	return challengeConfigs, nil
+}
+
+func (cc *ChallengesClient) GetPercentiles() (map[string]map[string]float64, error) {
+	logger := cc.logger().WithField("method", "GetPercentiles")
+	var percentiles map[string]map[string]float64
+	if err := cc.c.GetInto(endpointChallangesPercentiles, &percentiles); err != nil {
+		logger.Debug(err)
+		return nil, err
+	}
+	return percentiles, nil
+}
+
+func (cc *ChallengesClient) GetConfigWithChallengeId(challengeId int64) (*ChallengeConfigInfoDto, error) {
```

Comment:
In general, use `By...` instead of `With...` for names of methods with additional parameters. If there is only one method of the kind, there is no need to add the parameter in the method name at all.
```suggestion
func (cc *ChallengesClient) GetConfig(challengeId int64) (*ChallengeConfigInfoDto, error) {
```